cpp/src/Ice/EndpointI.h

```diff
@@ -111,7 +101,10 @@ class ICE_API EndpointI : public Ice::Endpoint
     // returning connectors sorted according to the endpoint selection
     // type.
     //
-    virtual void connectors_async(Ice::EndpointSelectionType, const EndpointI_connectorsPtr&) const = 0;
+    virtual void connectorsAsync(
+        Ice::EndpointSelectionType,
+        std::function<void(const std::vector<ConnectorPtr>&)>,
```

Comment:
We're passing a const& and not a copy because the caller (implementation of connectorsAsync) keeps a cached copy?

Reply:
As far as I can tell, the implementation always creates a new vector so we should be consistent with our async mapping and return a copy.

Reply:
Updated to pass by value